### PR TITLE
Only include participating drivers in calculate_driver_ahead

### PR DESCRIPTION
**Pull request by bambz96**
In the calculate_driver_ahead function, it now only calculates distance to driver ahead against drivers who participated in the session (i.e. drv_laps not empty for that driver)

### CODE REVIEW

File: fastf1/core.py [hunk 1]
```diff
@@ -894,37 +894,38 @@ def calculate_driver_ahead(self):
...
-            if not drv_laps_before.empty:
-                lap_n_before = drv_laps_before['LapNumber'].iloc[-1]
-                if lap_n_before < first_lap_number:
-                    # driver is behind on track an therefore will cross the finish line AFTER self
-                    # therefore above check for LapStartTime <= t_start is wrong
-                    # the first relevant lap is the first lap with LapStartTime > t_start which is lap_n_before += 1
-                    lap_n_before += 1
-            else:
-                lap_n_before = min(drv_laps['LapNumber'])
+            if not drv_laps.empty: # Only include drivers who participated in this session
```
Comment (theOehrly) [inline fastf1/core.py:897]: I'd prefer to have 
```
if drv_laps.empty:
    continue
```
The line lengths are getting a bit out of hand if you indent all the following code.

**Comment by theOehrly** [comment]
In general, this looks like the right way to fix this. But see the note in the review.

The flake8 lint test will run correctly and give you useful feedback if you rebase onto the current master. There was an issue with dependencies being incompatible.

**Comment by bambz96** [comment]
Thanks for the feedback, I've pulled from the current master and made the change you suggested.

File: fastf1/core.py [hunk 2]
```diff
@@ -894,6 +894,8 @@ def calculate_driver_ahead(self):
         for drv in self.session.drivers:
             # find correct first relevant lap; very important for correct zero point in distance
             drv_laps = self.session.laps[self.session.laps['DriverNumber'] == drv]
+            if not drv_laps.empty: # Only include drivers who participated in this session
```
Comment (theOehrly) [inline fastf1/core.py:897]: There's one single space missing before the inline comment. There need to be two spaces... sorry

Comment (bambz96) [reply fastf1/core.py]: No problem, I've just added it.

Comment (theOehrly) [reply fastf1/core.py]: I just saw this now when I wanted to merge it, the logic is wrong, The `not` shouldn't be there if we use continue.

Comment (bambz96) [reply fastf1/core.py]: Oops my bad...fixed now

**Comment by theOehrly** [comment]
Great, now everything looks good, thanks

**Pull request merged**